nautobot/extras/utils.py

```diff
@@ -496,6 +497,19 @@ def remove_prefix_from_cf_key(field_name):
     return field_name[3:]
 
 
+def check_if_key_is_graphql_safe(key):
+    """
+    Helper method to check if a key field is Python/GraphQL safe.
+    Used in CustomField for now, should be used in ComputedField and Relationship as well.
+    """
+    if key[0].isdigit():
+        raise ValidationError({"key": "The first letter of the CustomField key cannot be a number"})
+    if " " in key:
+        raise ValidationError({"key": 'There cannot be whitespaces " " in CustomField key'})
+    if "-" in key:
+        raise ValidationError({"key": 'There cannot be hypens "-" in CustomField key'})
```

Comment:
Probably should be more stringent still - per https://graphql-rules.com/rules/naming, the key should match the regexp `[_A-Za-z][_0-9A-Za-z]*`

Reply:
The latest implementation outputs the following `ValidationError` when there is a whitespace in the `key`:
```
>>> cf = CustomField.objects.create(label="cf 1", key="123 2")
>>> cf.validated_save()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/source/nautobot/core/models/__init__.py", line 51, in validated_save
    self.full_clean()
  File "/usr/local/lib/python3.10/site-packages/django/db/models/base.py", line 1251, in full_clean
    raise ValidationError(errors)
django.core.exceptions.ValidationError: {'key': ['Enter a valid “slug” consisting of letters, numbers, underscores or hyphens.'], "CustomField's specific key attribute": ['This key is not Python/GraphQL safe. Please do not start the key with a digit and do not use hypens or whitespaces']}
>>> 
```

It outputs two exception messages. The first one is emitted from Django's `SlugField` and the second one is our custom message. As you can see, these two messages are contradictory regarding hyphen usage. Wonder if is there an easy way to resolve this.